crates/astria-sequencer-relayer/src/data_availability.rs

```diff
@@ -154,75 +157,126 @@ struct RollupNamespaceData {
 
+#[derive(Debug)]
 pub struct CelestiaClientBuilder {
-    endpoint: String,
-    gas_limit: Option<u64>,
+    endpoint: Option<String>,
+    bearer_token: Option<String>,
+    gas_limit: u64,
+    fee: u128,
+}
+
+impl Default for CelestiaClientBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
 impl CelestiaClientBuilder {
-    pub fn new(endpoint: String) -> Self {
+    /// Create a celestia client builder with its fields default initialized.
+    pub(crate) fn new() -> Self {
         Self {
-            endpoint,
-            gas_limit: None,
+            endpoint: None,
+            bearer_token: None,
+            gas_limit: DEFAULT_PFD_GAS_LIMIT,
+            fee: DEFAULT_PFD_FEE,
         }
     }
 
-    pub fn build(self) -> eyre::Result<CelestiaClient> {
-        let cnc = CelestiaNodeClient::builder()
-            .base_url(self.endpoint)
-            .wrap_err("failed to set base URL for celestia node client; bad URL?")?
-            .build()
-            .wrap_err("failed creating celestia node client")?;
-        Ok(CelestiaClient {
-            client: cnc,
-            gas_limit: self.gas_limit.unwrap_or(DEFAULT_PFD_GAS_LIMIT),
-        })
+    pub fn endpoint(self, endpoint: &str) -> Self {
+        Self {
+            endpoint: Some(endpoint.to_string()),
+            ..self
+        }
+    }
+
+    pub fn bearer_token(self, bearer_token: &str) -> Self {
+        Self {
+            bearer_token: Some(bearer_token.to_string()),
+            ..self
+        }
     }
 
-    /// sets the gas limit to be used for PFDs.
     pub fn gas_limit(self, gas_limit: u64) -> Self {
         Self {
-            gas_limit: Some(gas_limit),
+            gas_limit,
+            ..self
+        }
+    }
+
+    pub fn fee(self, fee: u128) -> Self {
+        Self {
+            fee,
             ..self
         }
     }
+
+    pub fn build(self) -> eyre::Result<CelestiaClient> {
+        let Self {
+            endpoint,
+            bearer_token,
+            gas_limit,
+            fee,
+        } = self;
+        let client = {
+            Client::builder()
+                .set_endpoint(endpoint)
+                .set_bearer_token(bearer_token)
+                .build()
+                .wrap_err("failed constructing a celestia jsonrpc client")?
+        };
+        Ok(CelestiaClient {
+            client,
+            gas_limit,
+            fee,
+        })
+    }
 }
 
 /// CelestiaClient is a DataAvailabilityClient that submits blocks to a Celestia Node.
 pub struct CelestiaClient {
-    client: CelestiaNodeClient,
+    client: Client,
     gas_limit: u64,
+    fee: u128,
 }
 
 impl CelestiaClient {
+    pub fn builder() -> CelestiaClientBuilder {
+        CelestiaClientBuilder::new()
+    }
+
     pub async fn get_latest_height(&self) -> eyre::Result<u64> {
+        // request the network head and extract the `height` from it.
+        // NOTE: the previous gateway based client made a call to
```

Comment:
Is this comment mostly intended for code review? It feels like more context than needed, the function is `get_lastest_height`, we get it off the header block. Seems straight forward and logical.

Potentially a note about network vs local might be useful here though?

Reply:
Will remove the comment. It was intended for myself mainly.